plugins/Transitions/Transitions.php

```diff
@@ -55,4 +59,24 @@ public function getClientSideTranslationKeys(&$translationKeys)
         $translationKeys[] = 'CoreHome_ThereIsNoDataForThisReport';
         $translationKeys[] = 'General_Others';
     }
+
+    public function addJsGlobalVariables(&$out)
+    {
+
+        $idSite = Common::getRequestVar('idSite', 1, 'int');
+        $period = Common::getRequestVar('period');
+        $date = Common::getRequestVar('date');
+
+        $api = API::getInstance();
+        if($api->getPeriodAllowed($period, $idSite, $date)) {
+            $allowed = 'true';
+        } else {
+            $allowed = 'false';
+        }
+
+        $out .= "      
+        piwik.transitionsPeriodAllowed = $allowed;\n
```

Comment:
This won't work properly. I guess you will actually need to pass over the configured allowed period to javascript and do the check in javascript as well.
When switching the period/date in the date selector the page won't be fully reloaded, and so the global variables won't be refreshed. So it currently always uses the initial state matomo was opened with.

Reply:
Good point :+1:  I've changed it to pass over the max period allowed config setting and replicate the API check in Javascript. The transitions row action is now disabled/enabled when report dates are changed (without reloading)